pkg/plugin/resource/resource.go

```diff
@@ -0,0 +1,200 @@
+package resource
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"sort"
+	"text/template"
+
+	log "github.com/Sirupsen/logrus"
+	plugin_group "github.com/docker/infrakit/pkg/plugin/group"
+	"github.com/docker/infrakit/pkg/plugin/group/types"
+	"github.com/docker/infrakit/pkg/spi/group"
+	"github.com/docker/infrakit/pkg/spi/instance"
+	"github.com/twmb/algoimpl/go/graph"
+)
+
+const (
+	resourceGroupTag = "infrakit.resource-group"
+	resourceNameTag  = "infrakit.resource-name"
+)
+
+// NewResourcePlugin creates a new resource plugin.
+func NewResourcePlugin(instancePlugins plugin_group.InstancePluginLookup) group.Plugin {
+	return &plugin{
+		instancePlugins: instancePlugins,
+	}
+}
+
+// Spec is the configuration schema for this plugin, provided in group.Spec.Properties.
+type Spec struct {
+	Resources map[string]types.InstancePlugin
+}
+
+type resource struct {
+	plugin instance.Plugin
+	config types.InstancePlugin
+}
+
+type plugin struct {
+	instancePlugins plugin_group.InstancePluginLookup
+}
+
+func (p *plugin) CommitGroup(config group.Spec, pretend bool) (string, error) {
```

Comment:
Can you add a couple of tests for this with some example input?   In particular, the provisioning order (dependency) code can use a few tests to show usage and get some test coverage.  Thanks!